%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template
%%It must be present to enable the BMC Submission system to
%%recognise this template!!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%          <8 June 2012>              %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
% \documentclass{bmcart}

%%% Load packages
\usepackage{amsthm,amsmath}
\RequirePackage{natbib}
% \RequirePackage[authoryear]{natbib}% uncomment this for author-year bibliography
\RequirePackage[hypertexnames=false]{hyperref} % For urls and hyperlinks
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails

\usepackage[
%  disable, %turn off todonotes
  colorinlistoftodos, %enable a coloured square in the list of todos
  textwidth=2cm, %set the width of the todonotes
  textsize=scriptsize, %size of the text in the todonotes
  ]{todonotes}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %%
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %%
%%  submitted article.                         %%
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\def\includegraphic{}
\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs  %%% START

\newcommand{\hemaClass}{\href{http://hemaClass.org}{\texttt{hemaClass.org}}}
\newcommand{\R}{\textsf{R}}
\newcommand{\pkg}[1]{\textbf{#1}}

\DeclareMathOperator*{\median}{median}
\DeclareMathOperator*{\std}{std}

\endlocaldefs %% END


%%% Begin ...

\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{hemaClass.org: Online one-by-one microarray normalization and classification of hematological cancers for precision medicine}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   noteref={n1}                       % id's of article notes, if any
]{\inits{SF}\fnm{Steffen} \snm{Falgreen}}
\author[
   addressref={aff1,aff2},                   % id's of addresses, e.g. {aff1,aff2}
   noteref={n1}                        % id's of article notes, if any
]{\inits{AEB}\fnm{Anders E} \snm{Bilgrau}}
\author[
   addressref={aff1,aff2}                   % id's of addresses, e.g. {aff1,aff2}
]{\inits{LHJ}\fnm{Lasse H} \snm{Jakobsen}}
\author[
   addressref={aff1,aff2}                   % id's of addresses, e.g. {aff1,aff2}
]{\inits{JH}\fnm{Jonas} \snm{Have}}
\author[
   addressref={aff1,aff2}                   % id's of addresses, e.g. {aff1,aff2}
]{\inits{KLN}\fnm{Kasper L} \snm{Nielsen}}
\author[
   addressref={aff1}                   % id's of addresses, e.g. {aff1,aff2}
]{\inits{TCEG}\fnm{Tarec C} \snm{El-Galaly}}
\author[
   addressref={aff1}                  % id's of addresses, e.g. {aff1,aff2}
]{\inits{JSB}\fnm{Julie S} \snm{Bødker}}
\author[
   addressref={aff1}                  % id's of addresses, e.g. {aff1,aff2}
]{\inits{AS}\fnm{Alexander} \snm{Schmitz}}
\author[
   addressref={aff3}                  % id's of addresses, e.g. {aff1,aff2}
]{\inits{KHY}\fnm{Ken H} \snm{Young}}
\author[
   addressref={aff1,aff2}                  % id's of addresses, e.g. {aff1,aff2}
]{\inits{HEJ}\fnm{Hans E} \snm{Johnsen}}
\author[
   addressref={aff1,aff2}                 % id's of addresses, e.g. {aff1,aff2}
]{\inits{KD}\fnm{Karen} \snm{Dybkær}}
\author[
   addressref={aff1,aff2},
   corref={aff1},                       % id of corresponding address, if any
   email={mboegsted@dcm.aau.dk}
]{\inits{MB}\fnm{Martin} \snm{Bøgsted}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgname{Department of Haematology, Aalborg University Hospital}, % university, etc
  \street{Søndre Skovvej 15},                     %
  \postcode{9000}                                % post or zip code
  \city{Aalborg},                              % city
  \cny{DK}                                    % country
}
\address[id=aff2]{%
  \orgname{Department of Clinical Medicine, Aalborg University},
  \street{Søndre Skovvej 15},                     %
  \postcode{9000}                                % post or zip code
  \city{Aalborg},                              % city
  \cny{DK}                                    % country
}
\address[id=aff3]{%
  \orgname{Department of Hematopathology, University of Texas, MD Anderson Cancer Center},
  \street{1515 Holcombe Blvd},                     %
  \postcode{77030}                                % post or zip code
  \city{Houston, TX},                              % city
  \cny{US}
}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
% \note{http://hemaclass.org}     % note to the article
\note[id=n1]{Shared first authorship} % note, connected to author
\end{artnotes}

% \end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract
\parttitle{Background} %if any
Dozens of genomics based cancer classification systems have been introduced with prognostic, diagnostic, and predictive capabilities.
However, they often employ complex algorithms and are only applicable on whole cohorts of patients, making them difficult to apply in a personalized clinical setting.
\\\parttitle{Results}
This prompted us to create \hemaClass{}, an online web application providing an easy interface to one-by-one microarray based risk classifications of diffuse large B-cell lymphoma (DLBCL) and multiple myeloma (MM) into oncogenomic cell-of-origin and chemotherapeutic sensitivity classes.
As laboratory specific effects cannot be accounted for in true one-by-one microarray normalization, \hemaClass{} optionally allows the user to supply a reference dataset to increase the accuracy of the classifications. Classification results for one-by-one array pre-processing with and without a user supplied reference dataset were compared to cohort based classifiers in $4$ publicly available datasets.
Overall, \hemaClass{} yields satisfactory inter-method agreements across all datasets.
The website is essentially the \R{}-package \pkg{hemaClass} accompanied by a Shiny web application.
The well-documented package can be used to run the website locally or to use the developed methods programatically.
\parttitle{Conclusions}
The website and \R{}-package is relevant for biological and clinical lymphoma researchers as it provides reliable and swift methods for calculation of disease subclasses.
The proposed one-by-one pre-processing methods are relevant for all researchers using microarrays.
\medskip\\
\textbf{Keywords:~}
Diffuse Large B-Cell Lymphoma; Multiple Myeloma; Oncogenomics; Classification; Chemosensitivity; Microarray; Pre-processing; Robust Multichip Average; Gene expression profiling; Cell of origin
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{Diffuse Large B-Cell Lymphoma}
\kwd{Multiple Myeloma}
\kwd{Oncogenomics}
\kwd{Classification}
\kwd{Chemoresistance}
\kwd{Microarray}
\kwd{Pre-processing}
\kwd{Robust Multichip Average}
\kwd{Gene expression profiling}
\kwd{Cell of origin}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
\end{fmbox}% uncomment this for twcolumn layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%
%% Background %%
%%


\section{Background}
In addition to current clinically used risk factor scoring systems several independent gene expression profile (GEP) based risk stratifications have been proposed with biological and clinical significance in hematological cancers.
Although drug targetable genes which are only expressed in subtypes of the e.g.\ DLBCL tumours have been identified, they are not readily applicable in clinical research and routine settings due to lack of available routine diagnostic tests \cite{Jaffe2009, Sehn2014}.

Alizadeh et al.~\cite{Alizadeh2000} developed an important example of a biological sub-classification of lymphoma.
On the basis of GEP analyses, DLBCL cases were assigned to categories of activated B-cell phenotype (ABC) or germinal center B-cell phenotype (GCB) with different clinical outcomes.
The validity of this classification and its prognostic importance has been confirmed in a number of later studies \cite{Rosenwald2002a,Hans2004,Poulsen2005,Lenz2008a,Monti2012a}.
Recently, we have refined the ABC/GCB subclassification of DLBCL to include a B-cell Associated Gene Signature (BAGS) classifier \cite{DybkaerBoegsted2015} capable of classifying DLBCL samples into $5$ different B-cell subtypes:
Naive (N), Centrocyte (CC), Centroblast (CB), Memory (M), and Plasmablasts (PB).
For example, the BAGS classifier stratifies the GCB subclass into a CC subtype with superior survival as compared to the CB subtype \cite{DybkaerBoegsted2015} which opens up for considering different treatment regimes in the GCB class of patients.
In another study \cite{Falgreen2015} we developed classification based sensitivity gene signatures (REGS) for the most prominent drugs used in the treatment of DLBCL patients:
Cyclophosphamide (C), Doxorubicin (H), and Vincristine (O).
However, these and most existing algorithms are only applicable in the presence of whole cohorts of patients making them difficult to apply in a routine clinical setting.

The traditional lymphoma staging and risk classification systems are based on the Ann Arbor classification for staging of lymphoma (extent of disease and extranodal involvement) and simple prognostic tools such as the international prognostic index (IPI, \cite{IPI}) for large cell lymphoma and the Follicular Lymphoma International Prognostic Index (FLIPI, \cite{FLIPI}), both derived from patient age, performance status, easy available blood tests, and disease stage.
Due to the simplicity of these clinical risk stratification algorithms they are still the most widely used risk scoring systems today.
Risk stratification according to these algorithms have been systematised and made easily accessible for desktop, online, and even smart-phone use.
However, easily accessible molecular classification methods are lacking behind and thereby delaying the translation of new molecular findings into clinical practice.
A few methods exist for cancer types other than lymphoma, including acute myeloid leukemia (AML) \cite{Huang2009}.
But to the best of our knowledge, quite surprisingly, no tools have been developed for use in the lymphoma field despite very early promising results in the molecular biomarker field.
An accessible Windows based application has been published for ABC/GCB classification \cite{Care2013}.
Although Care et al.~\cite{Care2013} develops a new ABC/GCB classifier which is stable across microarray technologies and trial centres, the ABC/GCB classifier is potentially biased towards classes which differentiate the prognosis and not biology.
This bias arises as the ensemble of classifiers are chosen on basis of their ability to separate survival.

In clinical settings, the methods need to be applicable for a single patient sample and straightforward to use.
This prompted us to develop an easy accessible and flexible web-based tool for ABC/GCB, BAGS, and REGS classification which is capable of computing our recently developed classifiers for microarray data based on the Affymetrix's Human Genome U-133 Plus 2 arrays.
The classifications made by the web-based tool \hemaClass{} are compared to the existing state-of-the-art and approved ABC/GCB classifications of DLBCL.
We believe that \hemaClass{} will provide a novel and user-friendly concept for bringing complex molecular classification of disease more swiftly into daily clinical practice.
As already delineated, this paper primarily describes developed classification schemes for DLBCL although MM classification is currently also featured on the website.
A more extensive treatment of MM classification schemes is in preparation and will be implemented on \hemaClass{}.


\section{Implementation}


\subsection{Classification workflow}
The analysis workflow of \hemaClass{} can be divided into four parts:
(1) Three reference datasets which have been pre-processed and prepared offline are loaded upon server start,
(2) a graphical user interface with upload and input facilities,
(3) a server handling the online processing and classification of the user input, and
(4) the resulting classifications available for download and inspection via the user interface.
The workflow architecture is illustrated in Figure \ref{fig:webtooldiagram}.



\subsection{Software availability and technical details}
The interactive web application available at \url{hemaclass.org} was created using the statistical programming language \R{} \cite{RCoreTeam}, the software package \pkg{shiny} \cite{shiny}, and the accompanying Linux server software.
All \hemaClass{} functionality, including the RMA normalization and classification procedures, are available through the accompanying package \pkg{hemaClass} based on a number of packages from the Comprehensive R Archive Network \cite{RCoreTeam} and the Bioconductor environment \cite{Gentleman2004}.
The Shiny server handles the interaction between the front end web application and the back end \R{} processing.
The back end is essentially the well-documented \pkg{hemaClass} package which can be utilized as a programmatical interface to the functionality of the website.
However, the package also allows users to run a local instance of the website if one wishes to avoid uploading large files to our servers.
The development and latest version of \pkg{hemaClass} is open source and freely available at \url{https://github.com/oncoclass/hemaclass} for sharing, modification, and redistribution.
All bug-reports, suggestions, and comments on the website or package are welcome following the latter link.
The regular RMA pre-processing is carried out with the Bioconductor package \pkg{affy} \cite{Gautier2004}.
Core functions for the one-by-one and reference based RMA pre-processing are written in \textsf{C++} and imported to \R{} using \pkg{Rcpp} and \pkg{RcppArmadillo} \cite{Rcpp2013,Eddelbuettel2011,RcppArmadillo,Sanderson2010}.



\subsection{Data overview}
The seven gene expression datasets used in this paper are summarized in Table \ref{table:01}.
All GEP data are on the Affymetrix GeneChip HG-U133 Plus 2.0 array and available at the Gene Expression Omnibus (GEO) \cite{Barrett2013} website (\url{http://www.ncbi.nlm.nih.gov/geo/}).
To establish the classifiers the following datasets are used:
\begin{enumerate}
  \item Gene expression data from $181$ CHOP treated DLBCL patients are used to establish the ABC/GCB classifier.
  This cohort will be referred to as the \emph{LLMPP CHOP} (Lymphoma/Leukemia Molecular Profiling Project CHOP) cohort \cite{Lenz2008a}.
  The cohort is also used as a default reference set throughout the paper for normalization of arrays.
  \item The BAGS classifier is based on gene expression data from eight human tonsils sorted in five B-cell subsets.
  This dataset will be referred to as the \emph{tonsil dataset} and it is also used for scaling of gene expression data for BAGS classification \cite{DybkaerBoegsted2015}.
  \item The REGS classifiers are based on a panel of $12$ Multiple Myeloma (MM) and $14$ DLBCL cell lines.
  This panel will be referred to as \emph{BCELL26}.
  The DLBCL part of the cell line panel is used for scaling of patient data and this part of the dataset will be referred to as   \emph{DLBCL14} \cite{Falgreen2015}.
\end{enumerate}
For validation the following four DLBCL cohorts are used:
\begin{enumerate}
  \item[4.] The Aalborg OCT cohort (\emph{CHEPRETRO}) of $89$ Danish DLBCL patients undergoing first-line treatment at Aalborg University Hospital \cite{DybkaerBoegsted2015}.
  The clinical dataset is as registered in the National Clinical Quality Database for Malignant Lymphoma (LYFO, \url{lymphoma.dk}) \cite{Gang2012}.
  \item[5.] The International DLBCL Rituximab-CHOP Consortium MD Anderson (\emph{IDRC}) cohort of $470$ DLBCL patients treated with R-CHOP first-line therapy \cite{Visco2012}.
  Note, that these samples are formalin-fixed, paraffin-embedded (FFPE).
  \item[6.] The Lymphoma/Leukemia Molecular Profiling Project R-CHOP (\emph{LLMPP R-CHOP}) cohort of $233$ DLBCL patients treated with R-CHOP first-line therapy \cite{Lenz2008a}.
  \item[7.] The Mayo-Dana-Farber Cancer Institute (\emph{MDFCI}) cohort of $90$ DLBCL patients treated with R-CHOP first-line therapy \cite{Monti2012a}.
\end{enumerate}
The GEO datasets were downloaded using the \R{}-package \pkg{DLBCLdata} \cite{DLBCLdata}.





\subsection{One-by-one RMA normalization}
Recall that robust multichip average (RMA) pre-processing consists of three steps in order:
(1) Background correction,
(2) quantile normalization, and
(3) summarization of probes to probe-sets \cite{Irizarry2003}.
Confer Bolstad \cite{Bolstad2004} for a comprehensive account on RMA.
Both the quantile normalization and summarization procedures of RMA are cohort based and hence need to be altered to facilitate a one-by-one RMA pre-processing scheme.
As quantile normalizer, the empirical cumulative distribution function (ECDF) of the mean of the sample quantiles of the RMA background corrected LLMPP CHOP reference data is used in place of the usually applied ECDF of the mean of the sample quantiles of the user supplied data \cite{Bolstad2003}.
To mimic the summarization procedure of RMA \cite{Irizarry2003b} the probe effects estimated by median polish for the LLMPP CHOP reference data is subtracted all probes of the user data.
The RMA pre-processed expression value for each probe-set is then estimated as the median of the associated probes.
Finally, before classification the median of each probe-set in the LLMPP CHOP is subtracted from the corresponding probe-set in the user data.
This ensures identical classification probabilities whether data is supplied as a cohort or one-by-one.

To accommodate laboratory and sample preparation specific effects different from the LLMPP CHOP reference data, \hemaClass{} allows the user to upload an alternative dataset which is to be used in place of LLMPP CHOP as reference.
In this paper the reference datasets consisted of $30$ randomly selected samples from a given study.
The three pre-processing methods will be referred to as \emph{cohort}, \emph{one-by-one}, and \emph{reference based} normalization.



\subsection{Classification methods}



\subsubsection{Elastic nets}
Logistic and multinomial regression are used in all classification methods available at \hemaClass{}.
However, in GEP experiments, the number of probe-sets present on the microarray always outnumbers the sample size.
Collinearity present among the features further aggravates the problem of identifying genes responsible for the underlying biological mechanism.
Regression under these ill-posed circumstances is typically handled by so-called regularisation.
Here the elastic net penalty \cite{Friedman2010, Zou2005} is used which is a combination of the lasso \cite{Tibshirani1996} and ridge regression \cite{Hoerl1970}.
Similar to the lasso, this penalty ensures simultaneous variable selection and model estimation by forcing small coefficients to be zero, yielding sparse solutions.
In contrast to the lasso, the elastic net penalty is capable of selecting more variables than samples.

The elastic net penalty contains two parameters $\alpha$ and $\lambda$.
The parameter $\alpha$ interpolates the elastic net penalty between the ridge and the lasso penalty which corresponds to values of $0$ and $1$, respectively.
The parameter $\lambda$ determines the amount of shrinkage of the coefficients with larger values inducing more shrinkage until no variables are contained in the model.
Regularised logistic and multinomial regression are performed with the \R{}-package \pkg{glmnet} \cite{Friedman2010}.



\subsubsection{ABC/GCB classification}
The ABC/GCB classifier is established using logistic regression with an elastic net penalty on the LLMPP CHOP cohort.
Of the $181$ patients $74$ were ABC, $76$ were GCB, and $31$ were non-classified.
Using the $150$ patients classified as either ABC or GCB the dichotomous classifier capable of assigning each sample an estimate of the probability of being ABC was established.

To avoid over-fitting and limit the number of noise contributing genes, the elastic net parameters $\alpha$ and $\lambda$ were chosen through $10$ fold cross-validation.
The parameter $\alpha$ was varied between $0.1$ and $1$ with step size $0.025$ and $\log(\lambda)$ was varied between $-10$ and $2$ with step size $0.06$.
The optimal combination of the parameters and thereby the number of probe-sets were found at the values minimising the deviance.
The results of the cross validations are shown in Supplementary Figure S1.%\ref{fig:crossval}.
The minimum $0.13$ was attained at $\alpha = 0.15$ and $\log(\lambda) = -7.41$ resulting in a gene expression classifier consisting of $381$ probe-sets.

When a tumour sample is classified according to the ABC/GCB classifier using one-by-one normalization the data is used directly when classifying. However, when a tumour sample is classified according to the ABC/GCB classifier using reference based normalization the associated gene expressions are rescaled probe-set wise by the standard deviation of the LLMPP CHOP data divided by the standard deviation of the user supplied reference data.



\subsubsection{REGS classification}
\label{sec:regsmethods}
In the paper by Falgreen et al.~\cite{Falgreen2015} REGS classifiers were established for prediction of sensitivity to the drugs C, H, and O.
The classifiers were established on BCELL26 using regularised logistic regression analogous to the procedure described for the aforementioned ABC/GCB classifier.

The probability of sensitivity to the combination therapy $p_{CHO}$ is estimated based on the probabilities of drug sensitivity toward each of the three drugs $p_C$, $p_H$, and $p_O$, respectively.
This probability is calculated by a formula derived as the posterior probability of being sensitive given sensitivity towards each of the individual drugs under the assumption of conditional independence and uniform priors.
The formula is also known as Graham's formula,
\begin{equation*}
  p_{CHO} = \frac{p_C p_H p_O}{p_C p_H p_O + (1 - p_C)(1 - p_H)(1 - p_O)},
\end{equation*}
and derived in supplementary section S2.%\ref{sec:graham}.
If a drug is left out in the combination therapy the drug is simply removed from the formula.

When a tumour sample is classified according to the REGS classifiers the associated gene expressions are rescaled probe-set wise by the standard deviation of DLBCL14 divided by the standard deviation of the LLMPP CHOP dataset.
Similarly to the other classifiers the LLMPP CHOP cohort is replaced by the user reference dataset, if supplied.

Sensitvity classifiers for other chemotherapeutic drugs and diseases are also available on \hemaClass{} though established elsewhere \cite{Boegsted2011,Bogsted2013,Laursen2014,Laursen2015}.
The Rituximab sensitivity classifier \cite{Laursen2014, Laursen2015} is based upon an elastic net approach, like above, however using three classes.
The Melphalan sensitivity classifier \cite{Boegsted2011} uses sparse partial least squares to assign the classes sensitive, intermediate, and resistant to samples.
This classifier was developed for multiple myeloma (MM) patients and thus based on other data \cite{Boegsted2011}.
A classifier for dexamethasone is still considered experimental and currently available only in the package.



\subsubsection{BAGS classification}
The BAGS classifier established by Dybk{\ae}r et al.~\cite{DybkaerBoegsted2015} was based on multinomial regression regularised by an elastic net penalty.
The classifier was trained on the Tonsil dataset in a manner similar to the ABC/GCB classifier.

When a tumour sample is classified according to the BAGS classifier the associated gene expressions are rescaled probe-set wise by the standard deviation of the tonsil data divided by the standard deviation of the LLMPP CHOP dataset.
If the user supplies a reference dataset this is used in place of LLMPP CHOP.
The rescaling is performed to make the data comparable to the tonsil dataset.



\subsection{Inter-method reproducibility assessments}
To evaluate the reproducibility of the class probabilities obtained through cohort based normalization and \hemaClass{} normalization, Pearson's correlation coefficient for the logit-transformed probabilities and $95\%$ confidence interval (CI) were calculated for each classifier and dataset.
The identity and \emph{total} least square regression lines were compared to assess bias in the estimated probabilities \cite{CHEN1989}.
Total least squares regression was used as errors are present in both classification probabilities.

For each classifier the associated categories were obtained by thresholding the estimated probabilities.
The ABC/GCB classifier was thresholded by $0.1$ and $0.9$, i.e.\ a tumour sample was classified as ABC when the estimated probability exceeds $0.9$, GCB when it is beneath $0.1$, and unclassified otherwise.
For the BAGS classifier a tumour was classified as the class N, CB, CC, M, or PB with the highest probablity, if the associated probability exceeded $0.5$ and unclassified when this threshold was not met for any subtype.
For the REGS classifiers, C, H, O, and CHO combined, the threshold used were the $33\%$ and $66\%$ percentile of the estimated probabilities.
The classifiers were applied to datasets using the cohort, one-by-one, and reference based RMA normalization.
After thresholding the class probabilities confusion matrices were created.
From these the rate of agreement (or accuracy), Cohen's weighted $\kappa$, and corresponding $95\%$ CIs were computed to assess the agreement between the determined classes.


\section{Results}
\subsection{Using \hemaClass{}}
The website is an easy-to-use, interactive interface for \pkg{hemaClass} with the desired RMA normalization and the classification methods selected by the user.
The usage of the website is largely self-explanatory with context-dependent boxes aiding users with further information, warnings, or errors.
A comprehensive tutorial and guide to both the website and package is provided on the website or by running \texttt{vignette("howto")} in \R{}.
Users upload patient samples which are normalized and classified depending on settings chosen by the users.
A text summary of the results and a plot of the predicted progression free survival and overall survival is optionally provided for each patient sample.
These predictions are based on the patient survival data from LLMPP - RCHOP , MDFCI, and IDRC, as combined in \citet{DybkaerBoegsted2015}, using a Cox proportional hazard's regression with predictors IPI, at levels $0,1,2,4,5$, and a factor combining the ABC/GCB and BAGS classes, at levels ABC, GCB, GCB-CC, GCB-CB, and NC.



\subsection{ABC/GCB classification}
In order to classify patients as ABC/GCB based on the implemented one-by-one normalization method a classifier based on the regularised logistic regression was established.
The classifications were evaluated in the four clinical cohorts CHEPRETRO, MDFCI, IDRC, and LLMPP R-CHOP, which have all been classified according to the Wright's naive Bayes classifier \cite{Wright2003,Lenz2008a}.
The rate of agreement between the two classifiers are shown in Table \ref{tab:ABCGCBclassifier}.
Note that unclassified samples were included in the estimation of this rate i.e.\ a patient classified as ABC by one classifier but unclassified by the other is considered an error.
The table also includes the alternative measure of agreement, Cohen's weighted $\kappa$, were misclassifications involving the unclassified group are weighted by $1/2$.
The accompanying confusion matrices are shown in the first rows of Supplementary Table S1.%\ref{tab:confusionABCGCBHEMA}.

The logit probability of being ABC estimated using the established cohort-based classifier was compared to the corresponding estimate obtained through the one-by-one based classification scheme in Figure \ref{fig:ABCGCBDrug}A for CHEPRETRO.
The probabilities estimated through the two methods are very similar.
However, the probabilities estimated under one-by-one normalization are slightly uncalibrated (or biased) and skewed upwards, indicating that different cut points might be used for the classifications.
By normalizing to a reference set of $30$ randomly selected samples from CHEPRETRO this error and bias is greatly minimised as shown in Figure \ref{fig:ABCGCBDrug}B.

For both methods, patients are classified as ABC when the estimated probability exceeds $0.9$ and GCB when it is below $0.1$.
In Table \ref{tab:classALL} the resulting classifications are compared in terms of accuracy and Cohen's weighted $\kappa$ for one-by-one normalization and four validation datasets normalized to 30 internally sampled reference samples.
CHEPRETRO, MDFCI, IDRC, and LLMPP R-CHOP all show a high Cohen's weighted $\kappa$ and good rate of agreement, considering that misclassifications involving unclassified samples count as errors.
The reduced rate of agreement and Cohen's weighted $\kappa$ using one-by-one pre-processing in IDRC may be due to the samples being FFPE although this seems to be remedied by reference based pre-processing.
The accompanying confusion matrices are shown in the lower part of Supplementary Table S1.%\ref{tab:confusionABCGCBHEMA}.
Note, changes in predicted classes are mainly due to shifts into NC from ABC or GCB.
Direct disagreements between the classifiers are seemingly rare and only occurs in the IDRC dataset.


\subsection{REGS classification}
The probability of sensitivity towards each of the three drugs C, H, and O was estimated using \hemaClass{} both in terms of one-by-one and reference based normalization.
The logit probabilities of sensitivity are plotted against those obtained by cohort based normalization in Figure \ref{fig:ABCGCBDrug} for CHEPRETRO data.
Panels C, E, G, and I show the plots based on one-by-one normalization and panels D, F, H, and J show the plots for the reference based normalization.
The probabilities obtained by one-by-one and cohort based normalization are comparable. However, similar to the other classifiers the one-by-one normalization leads to slightly skewed and biased probabilities, indicating that different cut points should be considered.
The probabilities obtained by the reference based normalization resembles the cohort based to a great extent indicating that similar well-calibrated probabilities are obtainable for different laboratories by supplying a reference set.

Based on the estimated probabilities, the patients were categorised as sensitive, intermediate, or resistant based on the thresholds specified in Section \ref{sec:regsmethods}.
The classes obtained by \hemaClass{} are compared to those obtained by the cohort based approach in Table \ref{tab:classALL} in terms of rate of agreement and Cohen's weighted $\kappa$.
The associated confusion matrices for one-by-one normalization and reference based normalization are shown in Supplementary Table S3 %\ref{tab:confusiondrugonebyone}
and Supplementary Table S4, %\ref{tab:confusiondrugreference},
respectively.
Again, the reference based approach yields superior agreement with classifications using cohort based normalization.


\subsection{BAGS classification}
The BAGS classifier was evaluated in a manner similar to the ABC/GCB classifier.
The logit probability of a patient's tumour originating from one of the five subpopulations is again estimated by means of cohort, one-by-one, and reference based normalization.
The logit probabilities estimated by the one-by-one normalization is plotted against that of the cohort based normalization in Figure \ref{fig:Bagscorr} panels A, C, E, G, and I for CHEPRETRO.
The correlations between the flogit probabilities are highly significant, although slightly skewed and biased.
To accommodate this, the data was also normalized using a reference dataset.
The logit probabilities estimated by the reference based normalization is plotted against that estimated for the cohort based normalization in Figure \ref{fig:Bagscorr} panels B, D, F, H, and J for CHEPRETRO.
The reference based normalization removes much of the aforementioned bias.

Based on the probabilities estimated by means of each of the three normalization methods the patients of the four clinical cohorts are grouped into the BAGS.
The rate of agreement between the cohort based and \hemaClass{} classifications is shown in Table \ref{tab:classALL} along with the Cohen's weighted $\kappa$.
The associated confusion matrices are shown in Supplementary Table S2.%\ref{tab:BAGShemaclass}.




\section{Discussion}
Despite the enormous resources spent on developing molecular based cancer classification systems,
most of these are still not available in daily clinical practice.
% there seems to be an inertial resistance to develop clinical applications.
To allow for fast validation of our recent findings \cite{DybkaerBoegsted2015, Falgreen2015}, we have developed an easily accessible web application that permits other users to apply ABC/GCB, BAGS as well as drug sensitivity classification on their own datasets.

The one-by-one and referece based classifications performed by \hemaClass{} were compared to those obtained using cohort based normalization in four clinical cohorts.
Even though gene expression data for the four cohorts was obtained through varying preparation kits and sample storage methods, the results show that a one-by-one array analysis approach is feasible and performed comparably with the whole cohort based method.
It seems that this approach allows for a realistic application of microarray based lymphoma classification for research projects and, after suitable standardisation and calibration, even for clinical use.

The present treatment algorithms for DLBCL are based on disease stage and clinical risk stratification without accounting for underlying tumour-biology \cite{Schmoll2012} and does not routinely account for the enormous variations in tumor biology between patients.
The CHOP combination therapy (cyclophosphamide, doxorubicin, vincristine, and prednisone) has been the backbone of DLBCL therapy for decades with the only significant improvement being the addition of monoclonal CD20 antibodies (Rituximab) \cite{Coiffier2002a}.
Despite the addition of antibody therapy to conventional chemotherapy only $55\%$ of patients with poor risk disease achieve durable remission \cite{Ziepert2010}.
Thus, the need for new therapeutic options in DLBCL is obvious.
Currently a number of new drugs have shown promising activity in DLBCL, but their role outside clinical trials have not been defined.
These drugs are different from conventional chemotherapeutic compounds by targeting specific deregulated cell-cycle pathways \citep{Friedberg2011}.
An important example is inhibition of the NF-$\kappa$B pathway by proteasome inhibitors (i.e.\ bortezomib).
Interestingly, the constitutive activation of the NF-$\kappa$B pathway is characteristic for the ABC subtype of DLBCL which consequently enhances the effect of bortezomib in this subtype \citep{Dunleavy2009}.

With the increasing number of new drugs likely to become available over the next years and the fact that their efficacy may vary between subsets of patients defined by gene expression profiles, the current treatment of patients based on disease stage and clinical information alone will not be sufficient.
\hemaClass{} provides an example of fast processing of complex molecular information in a way that is simple and readily at hand for clinicians.

An immediate limitation of the study is the need for a reference dataset established under the same conditions as the samples one wishes to classify.
However, calibration of laboratory equipment is a well-known issue for many experimental techniques used in molecular biology like qPCR, mass spectrometry, immunohistochemistry, and flow cytometry.
An important part of the calibration is that samples should be calibrated towards a dataset consisting of a representative set of tissue samples.
For the microarray technology this could be solved by inventing a central tissue bank of e.g.\ B-cell cancer cell-lines of varying subtypes which could be used as reference set for each individual microarray platform.

The user reference dataset is important if one wish to use the conventionally used thresholds for classification.
However, without a user reference set, the high correlation coefficients demonstrate that equally high classification accuracies can be achieved if laboratory specific thresholds are carefully chosen.

Another limitation of the current web application is that it is only designed for one array type.
The server works at the moment only on Affymetrix HG-U133 Plus 2.0 microarrays, but extensions to other high throughput platforms and cancers are planned.
This can, however, be circumvented by either re-annotation to HUGO Gene Nomenclature Committee (HGNC) approved symbols as suggested by Care et al.~\cite{Care2013} or by re-annotated chip definition files as suggested by Dai et al.~\cite{Dai2005}.
At the moment we are working on extending the web application to other array types along the lines of Dai et al.~\cite{Dai2005}.

Traditionally ABC/GCB classification has been achieved using the naive Bayes classifier of Wright et al.~\cite{Wright2003} which is based on cohorts, MAS5.0 normalized arrays, and a Bayesian approach assuming an equal amount of ABC and GCB patients.
However, a classifier based on logistic regression regularised by an elastic net penalty was implemented to make the classification more adaptable to RMA normalization, one-by-one processing, and stable towards varying amounts of ABC/GCB patients.
This classifier proved to be quite comparable with the naive Bayes classifier over the four studied datasets confirming the strong and stable signal of the ABC/GBC subclasses of DLBCL.

Under the validation of the one-by-one method one should notice that the unclassified is treated as a class in its own right.
This implies a lower accuracy compared to an approach where the unclassified are left out of the validations.
The latter approach seems reasonable as changing classifications to unclassified is less serious than changing real classes.
Despite the disputed properties of Cohen's $\kappa$ the conservative approach is retained and the issue is addressed using a Cohen's weighted $\kappa$ approach.
Given that an idealised approach is problematic to formulate, readers are encouraged to consider the confusion matrices in the supplementary material to make an overall evaluation of the performance.

ABC/GCB, BAGS, and REGS are only a part of the GEP-based armamentarium of methods for stratifying lymphoma patients into risk groups \cite{Shipp2002, Lossos2004a, Malumbres2008} and it would be interesting to extend the tool to include other classification systems.
For a comprehensive review see \cite{Coutinho2013}.
To our knowledge only a few other classification methods from other cancer types have been made easily accessible as either web applications or desktop applications.
Hopefully, this research will inspire bioinformaticians and statisticians to make their cancer classification methods easily accessible for usage, speedy validation, critical reviews, and mutual inspiration.

\section{Conclusion}
Despite high throughput molecular biology has been around for almost two decades, only a few of the numerous biomarkers identified have undergone extensive validation and made it into the clinic \cite{Chen2012a}.
It is our hope that making our own findings publicly available in this way will speed up validation and testing of BAGS and REGS by other researchers without having to delve into extensive bioinformatics implementations.
Although \hemaClass{} is still separated from the clinic we believe a web based tool and suggestion for a clinical reference sample will bring cancer classification closer to the clinic.
Hopefully, this work can also spawn interesting discussions on the clinical requirements of GEP based diagnostic and prognostic tools.

All material for reproducing this paper and its results is found at \url{https://github.com/oncoclass/hemaclass-paper}.
Comments, suggest, bug reports, and other issues are warmly welcome at \url{https://github.com/oncoclass/hemaclass/issue} or by mail to the corresponding author.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Availability and requirements}
\begin{itemize}
  \item \textbf{Project name:} hemaClass
  \item \textbf{Project home page:} \url{http://hemaClass.org}
  \item \textbf{Operating system(s):} Platform independent
  \item \textbf{Programming language:} \textsf{R}, \textsf{C++}
  \item \textbf{Other requirements:} R 3.2.2 or higher.
  \item \textbf{License:} Apache License Version 2.0
  \item \textbf{Any restrictions to use by non-academics:} See license.
\end{itemize}

\section*{Competing interests}
The authors declare that they have no competing interests.

\section*{Authors' contributions}
SF, AEB, and MB concieved hemaClass.org and drafted the original manuscript.
SF and AEB created the hemaClass package and website.
SF, AEB, LHJ, JH, KLH and MB conducted the analyses and intepretation.
LHJ, JH, KLH contributed to the manuscript, package, and website.
TCEG, JSB, AS, HEJ, KD, and MB provided data and aided in interpreation of
the results.
All authors provided comments and approved the manuscript before submission.

\section*{Acknowledgements}
We thank Mads Boye and Bo Nygaard Bai at IT Services, Aalborg University, for their assistance on deploying the public server.
SF is supported by a Mobility PhD fellowship at the Graduate School of Health, Faculty of Health Sciences, Aarhus University.
The research is supported by MSCNET, a translational programme studying cancer stem cells in multiple myeloma supported by the EU FP6, CHEPRE, a programme studying chemo sensitivity in malignant lymphoma by genomic signatures supported by the Danish Agency for Science, Technology, and Innovation, and the National Experimental Therapy Partnership (NEXT), which is financed by a grant from Innovation Fund Denmark, as well as Karen Elise Jensen Fonden.
The technical assistance from Ann-Maria Jensen, Louise Hvilsh{\o}j Madsen, and Helle H{\o}holt is greatly appreciated.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{references}      % Bibliography file (usually '*.bib' )
% for author-year bibliography (bmc-mathphys or spbasic)
% a) write to bib file (bmc-mathphys only)
% @settings{label, options="nameyear"}
% b) uncomment next line
%\nocite{label}

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

% \section*{Figures}
%   \begin{figure}[h!]
%   \caption{\csentence{Sample figure title.}
%       A short description of the figure content
%       should go here.}
%       \end{figure}
%
% \begin{figure}[h!]
%   \caption{\csentence{Sample figure title.}
%       Figure legend text.}
%       \end{figure}


\begin{figure}[h!]
% \begin{center}
% \includegraphics[width=0.5\textwidth]{figures/Flowchart6.pdf}
% \end{center}
\caption{\csentence{Diagram of the workflow architecture.}}
\label{fig:webtooldiagram}
\end{figure}

\begin{figure}[h!]
% \begin{center}
% \includegraphics[width=0.375\textwidth]{figures/figure2.pdf}
% \end{center}
\caption{
\csentence{
Comparison of logit probabilities for the ABC/GCB and REGS classifiers obtained through one-by-one and reference based RMA normalization against cohort normalization.
}
The areas marked with green indicate patients with similar classification between cohort based normalization and one-by-one (A, C, E, G, I)  and reference based RMA normalization (B, D, F, H, J) .
Contrary, the areas marked with red indicate complete misclassifications.
For ABC/GCB and REGS the white areas indicate unclassified and intermediate sensitivity, respectively, in at least one of the classifiers.
The dashed and solid line shows the identity and total least squares line, respectively.}
\label{fig:ABCGCBDrug}
\end{figure}


\begin{figure}[h!]
% \begin{center}
% \includegraphics[width=0.375\textwidth]{figures/figure3.pdf}
% \end{center}
\caption{
\csentence{
Comparison of logit probabilities for the BAGS classifier obtained through one-by-one and reference based RMA normalization against cohort RMA normalization.
}
The coloured regions in the figure correspond to a threshold probability of $0.5$.
The dashed and solid line shows the identity and total least squares line, respectively.}
\label{fig:Bagscorr}
\end{figure}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%

\section*{Tables}
% \begin{table}[h!]
% \caption{Sample table title. This is where the description of the table should go.}
%       \begin{tabular}{cccc}
%         \hline
%            & B1  &B2   & B3\\ \hline
%         A1 & 0.1 & 0.2 & 0.3\\
%         A2 & ... & ..  & .\\
%         A3 & ..  & .   & .\\ \hline
%       \end{tabular}
% \end{table}

%%% TABLE 1
\begin{table}[!h]%[htb]
{\footnotesize
\caption{Overview of datasets and GEO accession numbers used as reference (Ref.) or validation (Val.) data.}
\label{table:01}%
\begin{center}
\begin{tabular}{rlclll}
\hline
No. & Dataset & $n$ & Usage & GEO no. & Ref.
\\
\hline
1. & LLMPP CHOP & 181 & Ref. & GSE10846 & \citep{Lenz2008a}
\\
2. & Tonsil & 33 & Ref. & GSE56315 & \citep{DybkaerBoegsted2015}
\\
3. & DLBCL14 & 14 & Ref. & - & \citep{Falgreen2015}
\\
4. & CHEPRETRO & 89 & Val. & GSE56315 & \citep{DybkaerBoegsted2015}
\\
5. & IDRC & 470 & Val. & GSE31312 & \citep{Visco2012}
\\
6. & LLMPP R-CHOP & 233 & Val. & GSE10846 & \citep{Lenz2008a}
\\
7. & MDFCI & 90 & Val. & GSE34171 & \citep{Monti2012a}\\
\hline
\end{tabular}
\end{center}
}
\end{table}


%% TABLE 2
\begin{table}[!h]%[!tbp]
{\small
\caption{Comparison of ABC/GCB classification performed using Wright's naive
Bayes classifier \citep{Wright2003} and the established elastic net classifier
based on cohort normalization
for both. The second column shows the rate of agreement (accuracy) between the
classifiers with $95\%$ CI. The third column shows the Cohen's weighted
$\kappa$ with $95\%$ CI.\label{tab:ABCGCBclassifier}}
\begin{center}
\begin{tabular}{lll}
\hline
\multicolumn{1}{l}{Dataset}&\multicolumn{1}{c}{Agreement}&\multicolumn{1}{c}{Cohen's $\kappa$}\tabularnewline
\hline
CHEPRETRO&$0.94~(0.87, 0.98)$&$0.94~(0.88, 1.00)$\tabularnewline
MDFCI&$0.78~(0.68, 0.86)$&$0.77~(0.66, 0.88)$\tabularnewline
IDRC&$0.82~(0.79, 0.86)$&$0.80~(0.76, 0.85)$\tabularnewline
LLMPP R-CHOP&$0.91~(0.86, 0.94)$&$0.90~(0.85, 0.96)$\tabularnewline
\hline
\end{tabular}\end{center}}

\end{table}


% TABLE 3
\begin{table*}[!h]%[!tbp]
{\footnotesize
\caption{Comparison of classifications obtained using cohort based
normalization and one-by-one normalization as well as reference based normalization.
The classifications are compared in terms of rate of argreement (accuracy),
Cohen's weighted $\kappa$, and Pearson's correlation coefficient $r$ all
supplied with $95\%$ CIs. The comparisons in the first and last three columns
are based on the one-by-one normalization method and the reference based
normalization method, respectively.\label{tab:classALL}}
\begin{center}
\begin{tabular}{llllclll}
\hline
\multicolumn{1}{l}{\bfseries }&\multicolumn{3}{c}{\bfseries One-by-one pre-processing}&\multicolumn{1}{c}{\bfseries }&\multicolumn{3}{c}{\bfseries Reference based pre-processing}\tabularnewline
\cline{2-4} \cline{6-8}
\multicolumn{1}{l}{}&\multicolumn{1}{c}{Agreement}&\multicolumn{1}{c}{Cohen's $\kappa$}&\multicolumn{1}{c}{Pearson's $r$}&\multicolumn{1}{c}{}&\multicolumn{1}{c}{Agreement}&\multicolumn{1}{c}{Cohen's $\kappa$}&\multicolumn{1}{c}{Pearson's $r$}\tabularnewline
\hline
{\bfseries ABC/GCB}&&&&&&&\tabularnewline
~~CHEPRETRO&$.88~(.79, .94)$&$.87~(.78, .97)$&$.999~(.998, .999)$&&$.98~(.91, 1.0)$&$.98~(.93, 1.0)$&$1.00~(.999, 1.00)$\tabularnewline
~~MDFCI&$.69~(.59, .78)$&$.68~(.53, .82)$&$.999~(.998, .999)$&&$.98~(.91, 1.0)$&$.98~(.85, 1.0)$&$1.00~(.999, 1.00)$\tabularnewline
~~IDRC&$.65~(.61, .69)$&$.62~(.57, .68)$&$.986~(.983, .988)$&&$.93~(.90, .95)$&$.93~(.90, .96)$&$.993~(.991, .994)$\tabularnewline
~~LLMPP R-CHOP&$.82~(.77, .87)$&$.82~(.74, .89)$&$.999~(.999, .999)$&&$.94~(.90, .97)$&$.94~(.90, .98)$&$.991~(.988, .993)$\tabularnewline
\hline
{\bfseries BAGS}&&&&&&&\tabularnewline
~~CHEPRETRO&$.58~(.47, .69)$&$.56~(.28, .84)$&-&&$.78~(.65, .88)$&$.74~(.33, 1.0)$&-\tabularnewline
~~MDFCI&$.54~(.43, .64)$&$.48~(.17, .79)$&-&&$.80~(.68, .89)$&$.83~(.30, 1.0)$&-\tabularnewline
~~IDRC&$.52~(.47, .56)$&$.41~(.32, .50)$&-&&$.79~(.75, .83)$&$.79~(.62, .96)$&-\tabularnewline
~~LLMPP R-CHOP&$.56~(.49, .62)$&$.53~(.36, .70)$&-&&$.88~(.82, .92)$&$.88~(.60, 1.0)$&-\tabularnewline
\hline
{\bfseries REGS}&&&&&&&\tabularnewline
~~CHEPRETRO&$.73~(.68, .78)$&$.71~(.64, .77)$&$.934~(.920, .946)$&&$.84~(.79, .88)$&$.83~(.76, .89)$&$.992~(.990, .994)$\tabularnewline
~~MDFCI&$.60~(.55, .65)$&$.55~(.48, .61)$&$.824~(.788, .855)$&&$.90~(.86, .94)$&$.89~(.83, .96)$&$.997~(.996, .997)$\tabularnewline
~~IDRC&$.52~(.49, .54)$&$.33~(.30, .36)$&$.660~(.635, .685)$&&$.85~(.84, .87)$&$.84~(.81, .86)$&$.981~(.979, .983)$\tabularnewline
~~LLMPP R-CHOP&$.58~(.54, .61)$&$.50~(.46, .54)$&$.810~(.786, .831)$&&$.90~(.87, .92)$&$.89~(.85, .92)$&$.992~(.990, .993)$\tabularnewline
\hline
\end{tabular}\end{center}}

\end{table*}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
  \subsection*{Supplementary}
    Additional file containing supplementary text, tables, and figures.


\end{backmatter}
\end{document}
